import-export-cli/integration/testutils/api_testUtils.go

```diff
@@ -455,33 +515,65 @@ func ValidateExportedAPIStructure(t *testing.T, args *ApiImportExportTestArgs) {
 
 	output, _ := exportAPI(t, args.Api.Name, args.Api.Version, args.ApiProvider.Username, args.SrcAPIM.GetEnvName())
 
-	ValidateAPIStructure(t, args.Api, output)
+	validateAPI(t, args.Api, output, args.IsDeployed, SampleAPIYamlFilePath)
+
+	assert.True(t, base.IsAPIArchiveExists(t, GetEnvAPIExportPath(args.SrcAPIM.GetEnvName()),
+		args.Api.Name, args.Api.Version))
+}
+
+func ValidateExportedAPIRevisionStructure(t *testing.T, args *ApiImportExportTestArgs) {
+	t.Helper()
+
+	// Setup apictl envs
+	base.SetupEnv(t, args.SrcAPIM.GetEnvName(), args.SrcAPIM.GetApimURL(), args.SrcAPIM.GetTokenURL())
+
+	// Export api from env 1
+	base.Login(t, args.SrcAPIM.GetEnvName(), args.CtlUser.Username, args.CtlUser.Password)
+
+	output, _ := exportAPIRevision(t, args.Api.Name, args.Api.Version, args.ApiProvider.Username, args.Revision, args.SrcAPIM.GetEnvName())
+
+	validateAPI(t, args.Api, output, args.IsDeployed, SampleRevisionedAPIYamlFilePath)
 
 	assert.True(t, base.IsAPIArchiveExists(t, GetEnvAPIExportPath(args.SrcAPIM.GetEnvName()),
 		args.Api.Name, args.Api.Version))
 }
 
-func ValidateAPIStructure(t *testing.T, api *apim.API, exportedOutput string) {
+func validateAPI(t *testing.T, api *apim.API, exportedOutput string, isDeployed bool, sampleFile string) {
 	// Unzip exported API
 	exportedPath := base.GetExportedPathFromOutput(exportedOutput)
 	relativePath := strings.ReplaceAll(exportedPath, ".zip", "")
 	base.Unzip(relativePath, exportedPath)
 
+	unzipedProjectPath := relativePath + string(os.PathSeparator) + api.Name + "-" + api.Version
+
 	// Read the api.yaml file in the exported directory
-	fileData, err := ioutil.ReadFile(relativePath + string(os.PathSeparator) + api.Name + "-" + api.Version + APIYamlFilePath)
+	fileData, err := ioutil.ReadFile(filepath.Join(unzipedProjectPath, APIYamlFilePath))
+
 	if err != nil {
 		t.Error(err)
 	}
+
+	validateAPIStructure(t, &fileData, sampleFile)
+	//validateAPIValues(t, &fileData, api)
```

Comment:
Any reason to have this commented code segment? Shall we remove it?

Reply:
Currently this is breaking when we try to compare value from REST API vs what is there in project. Need to do changes to the DTO at test level to make this possible. Currently this requires extensive changes so did not include this in current PR. 

I'll remove this for now, maybe I can come up with a better way of doing this

Reply:
Fixed with https://github.com/wso2/product-apim-tooling/pull/728/commits/0901d9dcfc9501ea1bac6682987f76eae977e297

Reply:
Sure. Thanks